packages/main/src/SegmentedButton.ts

```diff
@@ -260,17 +263,71 @@ class SegmentedButton extends UI5Element {
 		}
 	}
 
+	/**
+	 * Performs the layout for the SegmentedButton component.
+	 *
+	 * Calculates and sets the appropriate width for the component based on the inline style and/or CSS class.
+	 * Adjusts the responsiveness of the component based on the parent width.
+	 *
+	 * @returns {Promise<void>} A Promise that resolves once the layout is completed.
+	 * @private
+	 */
 	async _doLayout(): Promise<void> {
-		const itemsHaveWidth = this.widths && this.widths.some(itemWidth => itemWidth > 2); // 2 pixels added for rounding
+		const rootFontSize = parseFloat(getComputedStyle(document.documentElement).fontSize); // Gets the root font size in pixels in order to convert rem to pixels correctly
```

Comment:
It will be clear to:
parseInt(getComputedStyle(this.items[0]).minWidth) 

this will give directly minWidth in pixels, and you will avoid use of hardcoded constant (which can be theme-specific btw)